src/assignments/tech-jobs-oo.rst

```diff
@@ -0,0 +1,558 @@
+#. Use ``Assert`` statements to test that the constructor correctly assigns the
+   class and value of each field.
+
+   .. admonition:: Tip
+
+      The ``is`` keyword can be used to check the class of an object and cast the object to the appropriate type if possible.
+      The result of the comparison is a boolean. For a review on how the ``is`` keyword works, check out how it is used by :ref:`Visual Studio <equals-shortcut>` when overriding the ``Equals()`` method!
+
+      .. sourcecode:: csharp
+
+         objectName is ClassName VariableName
+
+Test the ``Equals()`` Method
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Two ``Job`` objects are considered equal if they have the same ``id`` value,
+even if one or more of the other fields differ. Similarly, the two objects
+are NOT equal if their ``Id`` values differ, even if all the other fields are
+identical.
+
+#. In ``JobTest``, define a test called ``TestJobsForEquality``.
+#. Generate two ``Job`` objects that have identical field values EXCEPT for
+   ``id``. Test that ``Equals()`` returns ``false``.
+
+It might seem logical to follow up the ``false`` case by testing to make sure
+that ``Equals()`` returns ``true`` when two objects have the same ID. However,
+the positive test is irrelevant in this case.
+
+The way you built your ``Job`` class, each ``id`` field gets assigned a unique
+value, and the class does not contain a setter for the ``id`` field. You also verified
+that each new object gets a different ID when you tested the constructors.
+Without modifying the constructors or adding a setter, there is no scenario in
+which two different jobs will have the same ID number. Thus, we can skip the
+test for this condition.
+
+.. admonition:: Tip
+
+   Time to save, commit, and push your work to GitHub again.
+
+Use TDD to Build The ``ToString()`` Method
+------------------------------------------
+
+To display the data for a particular ``Job`` object, you need to implement a
+custom ``ToString()`` method. Rather than creating this method and then testing
+it, you will flip that process using TDD.
+
+Create First Test for ``ToString()``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Before writing your first test, consider how we want the method to behave:
+
+#. When passed a ``Job`` object, it should return a string that contains a
+   blank line before and after the job information.
+#. The string should contain a label for each field, followed by the data
+   stored in that field. Each field should be on its own line.
+
+   ::
+
+      ID:  _______
+      Name: _______
+      Employer: _______
+      Location: _______
+      Position Type: _______
+      Core Competency: _______
+
+#. If a field is empty, the method should add, "Data not available" after
+   the label.
+#. (Bonus) If a ``Job`` object ONLY contains data for the ``Id`` field, the
+   method should return, "OOPS! This job does not seem to exist."
+
+In ``JobTests``, add a new test to check the first requirement, then run
```

Comment:
first requirement (item 1. above) or something like this

Reply:
Done!